### DISCUSSION TITLE
Using node-modules in user extensions?

### DISCUSSION THREAD

**Discussion by sjorge** [Koenkk/zigbee2mqtt#12966] [General]
I'm not really sure how to go about this, I'd like to write a user extension to submit some data to influxdb.
But it requires `@influxdata/influxdb-client`, however that's not available from within zigbee2mqtt.

I can install it but every update when `npm ci` is run it gets removed (as expected), how do other user extension users deal with additional module requirements?

**Comment by Koenkk** [comment]
@nurikk got a trick? 😄

**Comment by nurikk** [comment]
Hi, @sjorge! Have you tried to install influx client as global package? `npm install -g @influxdata/influxdb-client` ?

ps In case if this wouldn't work, I might have few more tricks in my sleeve :)

**Reply by sjorge** [reply]
Installing globally wasn't an option, but I think I found a solution. At least with a test module I can just drop it in

`ZIGBEE2MQTT_DATA/extension/node_modules`, now I just need to install the influxdb-client there.

**Reply by nurikk** [reply]
you can init npm package inside `ZIGBEE2MQTT_DATA/extension` folder and enjoy all benefits of using package.json for extensions :)
